pkg/querier/queryrange/index_stats_cache.go

```diff
@@ -50,18 +52,35 @@ func (p IndexStatsExtractor) ResponseWithoutHeaders(resp queryrangebase.Response
 
 type IndexStatsCacheConfig struct {
 	queryrangebase.ResultsCacheConfig `yaml:",inline"`
+	// NOTE: We cannot use something similar to the per-tenant `max_cache_freshness_per_query` limit
+	// because resultsCache.filterRecentExtents would extract already inflated stats.
+	// Instead, we need to filter out whole requests that ask for stats in recent data.
+	DoNotCacheRequestWithin time.Duration `yaml:"do_not_cache_request_within"`
```

Comment:
Can we make this a tenant config, `max_index_cache_freshness`?

Reply:
Done